components/tidb_query_shared_expr/src/encryption.rs

```diff
@@ -0,0 +1,52 @@
+use flate2::read::{ZlibDecoder, ZlibEncoder};
+use flate2::Compression;
+use std::io::Read;
+use tidb_query_datatype::codec::data_type::*;
+
+pub fn compress(input: &[u8]) -> Option<Bytes> {
```

Comment:
I think it better for the method and the `uncompress` method to return a `Result`.